tests/functional/test_utils.py

```diff
@@ -40,3 +43,24 @@ def test_can_load_token(self):
         loader = FileWebIdentityTokenLoader(self.token_file)
         token = loader()
         self.assertEqual(self.token, token)
+
+
+class TestInstanceMetadataFetcher(unittest.TestCase):
+    def test_catch_retryable_http_errors(self):
+        with mock.patch('botocore.httpsession.URLLib3Session.send') as send_mock:
+            fetcher = InstanceMetadataFetcher()
+            send_mock.side_effect = EndpointConnectionError(endpoint_url="foo")
```

Comment:
Ok a couple tweaks here now that I've played around with it. `EndpointConnectionError` is the only retyable exception that doesn't meet this criteria actually. This test will pass with or without the change in utils. Let's change the exception to `ConnectionClosedError` instead.